app/deliver_grant_funding/forms.py

```diff
@@ -1,16 +1,46 @@
 from flask_wtf import FlaskForm
-from govuk_frontend_wtf.wtforms_widgets import GovRadioInput, GovSubmitInput, GovTextArea, GovTextInput
+from govuk_frontend_wtf.wtforms_widgets import (
+    GovCharacterCount,
+    GovRadioInput,
+    GovSubmitInput,
+    GovTextArea,
+    GovTextInput,
+)
 from wtforms.fields.choices import RadioField
-from wtforms.fields.simple import StringField, SubmitField
-from wtforms.validators import DataRequired
+from wtforms.fields.simple import StringField, SubmitField, TextAreaField
+from wtforms.validators import DataRequired, Email, ValidationError
 
+from app.common.data.interfaces.grants import grant_name_exists
 from app.common.data.types import QuestionDataType
+from app.common.forms.validators import MaxWords
 
 
 def strip_string_if_not_empty(value: str) -> str | None:
     return value.strip() if value else value
 
 
+class UniqueGrantName:
+    """Validator to ensure grant name is unique."""
+
+    def __init__(self, message: str | None = None):
+        self.message = message or "Grant name already in use"
+
+    def __call__(self, form: FlaskForm, field: StringField) -> None:
+        if field.data and grant_name_exists(field.data):
+            raise ValidationError(self.message)
+
+
+class ConditionalGGISRequired:
+    """Validator to ensure GGIS number is provided when 'yes' is selected."""
+
+    def __init__(self, message: str | None = None):
+        self.message = message or "Enter your GGIS reference number"
+
+    def __call__(self, form: FlaskForm, field: StringField) -> None:
+        if form.has_ggis.data == "yes" and (not field.data or not field.data.strip()):
+            raise ValidationError(self.message)
+
```

Comment:
Let's do this on the specific form itself - as this isn't reusable across forms, it probes `has_ggis` directly which is not gonna exist always.